mono/utils/mono-log-windows.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * mono-log-windows.c: Simplistic simulation of a syslog logger for Windows
+ *
+ * This module contains the Windows syslog logger interface
+ *
+ * Author:
+ *    Neale Ferguson <neale@sinenomine.net>
+ *
+ */
+#include <config.h>
+
+#ifdef HAVE_UNISTD_H
+#include <unistd.h>
+#endif
+
+#ifdef HOST_WIN32
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <string.h>
+#include <glib.h>
+#include <errno.h>
+#include <time.h>
+#include "mono-logger.h"
+
+static FILE *logFile = NULL;
+static void *logUserData = NULL;
+static char *logFileName = L".//mono.log";
+
+/**
+ * mapSyslogLevel:
+ * 	
+ * 	@level - GLogLevelFlags value
+ * 	@returns The equivalent character identifier
+ */
+static inline char 
+mapLogFileLevel(GLogLevelFlags level) 
+{
+	if (level & G_LOG_LEVEL_ERROR)
+		return ('E');
+	if (level & G_LOG_LEVEL_CRITICAL)
+		return ('C');
+	if (level & G_LOG_LEVEL_WARNING)
+		return ('W');
+	if (level & G_LOG_LEVEL_MESSAGE)
+		return ('N');
+	if (level & G_LOG_LEVEL_INFO)
+		return ('I');
+	if (level & G_LOG_LEVEL_DEBUG)
+		return ('D');
+	return ('I');
+}
+
+/**
+ * mono_log_open_logfile
+ * 	
+ *	Open the logfile. If the path is not specified default to stdout. If the
+ *	open fails issue a warning and use stdout as the log file destination.
+ *
+ * 	@ident - Identifier: ignored
+ * 	@userData - Not used
+ */
+void
+mono_log_open_syslog(const char *ident, void *userData)
+{
+	logFile = fopen(logFileName, "w");
+	if (logFile == NULL) {
+		g_warning("opening of log file %s failed with %s",
+			  strerror(errno));
+	}
+	logUserData = userData;
+}
+
+/**
+ * mono_log_write_logfile
+ * 	
+ * 	Write data to the log file.
+ *
+ * 	@domain - Identifier string
+ * 	@level - Logging level flags
+ * 	@format - Printf format string
+ * 	@vargs - Variable argument list
+ */
+void
+mono_log_write_syslog(const char *domain, GLogLevelFlags level, mono_bool hdr, const char *format, va_list args)
+{
+	time_t t;
+	struct tm *tod;
+	char logTime[80],
+	      logMessage[512];
+	pid_t pid;
+	int iLog = 0;
+	size_t nLog;
+
+	if (logFile == NULL)
+		mono_log_open_logfile(NULL, NULL);
```

Comment:
Correct.

From: Johan Lorensson +ADw-notifications+AEA-github.com+ADw-mailto:notifications+AEA-github.com+AD4APg-
Reply-To: mono/mono +ADw-reply++0004b6fa5a67f79cc59b287326ef5d790b50c1c0788ad14492cf000000011369020f92a170ce03ea14AQA-reply.github.com+ADw-mailto:reply++0004b6fa5a67f79cc59b287326ef5d790b50c1c0788ad14492cf000000011369020f92a170ce03ea14AQA-reply.github.com+AD4APg-
Date: Friday, June 3, 2016 at 4:30 AM
To: mono/mono +ADw-mono+AEA-noreply.github.com+ADw-mailto:mono+AEA-noreply.github.com+AD4APg-
Cc: Neale Ferguson +ADw-neale+AEA-sinenomine.net+ADw-mailto:neale+AEA-sinenomine.net+AD4APg-, Mention +ADw-mention+AEA-noreply.github.com+ADw-mailto:mention+AEA-noreply.github.com+AD4APg-
Subject: Re: +AFs-mono/mono+AF0- Enhance log tracing (+ACM-2734)


In mono/utils/mono-log-windows.c+ADw-https://github.com/mono/mono/pull/2734+ACM-discussion+AF8-r65672874+AD4-:

+AD4-  +ACo-    +AEA-format - Printf format string
+AD4-  +ACo-    +AEA-vargs - Variable argument list
+AD4-  +ACo-/
+AD4- ++vo-
+AD4- ++moAXw-log+AF8-write+AF8-syslog(const char +ACo-domain, GLogLevelFlags level, mono+AF8-bool hdr, const char +ACo-format, va+AF8-list args)
+AD4- +AHs-
+AD4-       time+AF8-t t+ADs-
+AD4-       struct tm +ACo-tod+ADs-
+AD4-       char logTime+AFs-80+AF0-,
+AD4-             logMessage+AFs-512+AF0AOw-
+AD4-       pid+AF8-t pid+ADs-
+AD4-       int iLog +AD0- 0+ADs-
+AD4-       size+AF8-t nLog+ADs-
+AD4-
+AD4-       if (logFile +AD0APQ- NULL)
+AD4-               mono+AF8-log+AF8-open+AF8-logfile(NULL, NULL)+ADs-


mono+AF8-log+AF8-open+AF8-syslog ?

+IBQ-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub+ADw-https://github.com/mono/mono/pull/2734/files/583d1a2580a5fc61d0002920c17c23fa71375c3d+ACM-r65672874+AD4-, or mute the thread+ADw-https://github.com/notifications/unsubscribe/AAS2-rV9LNb44godrTOC8p2UsioZwtXIks5qH-YPgaJpZM4HsGtY+AD4-.